src/compiler/checker.ts

```diff
@@ -29632,6 +29632,18 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return isSomeSymbolAssignedWorker(rootDeclaration.name);
     }
 
+    function isSymbolAssignedInForStatementBody(symbol: Symbol) {
+        const forStatement = getRootDeclaration(symbol.valueDeclaration!).parent.parent;
+        Debug.assert(isForStatement(forStatement));
+        return !isPastLastAssignment(symbol, forStatement.statement);
+    }
+
+    function isUsedInForStatementBody(symbol: Symbol, location: Node) {
+        const forStatement = getRootDeclaration(symbol.valueDeclaration!).parent.parent;
+        Debug.assert(isForStatement(forStatement));
+        return location.pos >= forStatement.statement.pos && location.end <= forStatement.statement.end;
+    }

```

Comment:
I'd love to combine those 2 but I've failed so far to figure out an elegant function name for a function that would do both 😅